src/tests/dictionaryConcepts/actions/dictionaryConcept.test.js

```diff
@@ -217,6 +218,49 @@ describe('Test suite for dictionary concept actions', () => {
     });
   });
 
+  it('should fetch dictionary concepts by name', () => {
+    moxios.wait(() => {
+      const request = moxios.requests.mostRecent();
+      request.respondWith({
+        status: 200,
+        response: [concepts],
+      });
+    });
+
+    const expectedActions = [
+      { type: IS_FETCHING, payload: true },
+      { type: FETCH_DICTIONARY_CONCEPT, payload: [concepts] },
+      { type: IS_FETCHING, payload: false },
+    ];
+
+    const store = mockStore(mockConceptStore);
+
+    return store.dispatch(fetchConceptsByName('search')).then(() => {
+      expect(store.getActions()).toEqual(expectedActions);
+    });
+  });
+
+  it('should handle the error when search by name fails', () => {
+    moxios.wait(() => {
+      const request = moxios.requests.mostRecent();
+      request.respondWith({
+        status: 404,
+      });
+    });
+
+    const expectedActions = [
+      { type: IS_FETCHING, payload: true },
+      { type: IS_FETCHING, payload: false },
+    ];
+
+    const store = mockStore(mockConceptStore);
+
+    return store.dispatch(fetchConceptsByName('search')).then(() => {
+      expect(store.getActions()).toEqual(expectedActions);
```

Comment:
Can you explain how how above line expects that the the error is handled when search by name fails?

Reply:
@dkayiwa yes I can try to explain...  Just above that particular line there's this code
`    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 404,
      });
    });`
We are using a mocking library `moxios` for that... and when the status is mocked to be in any range `4xx`, then moxios will only run the sequence of reducer actions that are triggered in case the request fails and in this case they are 
`    const expectedActions = [
      { type: IS_FETCHING, payload: true },
      { type: IS_FETCHING, payload: false },
    ];`
First the loader shows... then because no data comes back (error), the next immediate action is to stop the loader.